Warning of uninitialised variables in tcp.c and udp.c

The issues were introduced in 4573edfa.

```
src/tcp.c:39:14: warning: variable 'flags' is used uninitialized whenever 'if' condition is
      false [-Wsometimes-uninitialized]
    else if (lua_isstring(L, 1)) {
             ^~~~~~~~~~~~~~~~~~
src/tcp.c:49:45: note: uninitialized use occurs here
    ret = uv_tcp_init_ex(ctx->loop, handle, flags);
                                            ^~~~~
src/tcp.c:39:10: note: remove the 'if' if its condition is always true
    else if (lua_isstring(L, 1)) {
         ^~~~~~~~~~~~~~~~~~~~~~~~
src/tcp.c:35:23: note: initialize the variable 'flags' to silence this warning
    unsigned int flags;
                      ^
                       = 0

src/udp.c:38:14: warning: variable 'flags' is used uninitialized whenever 'if' condition is
      false [-Wsometimes-uninitialized]
    else if (lua_isstring(L, 1)) {
             ^~~~~~~~~~~~~~~~~~
src/udp.c:48:45: note: uninitialized use occurs here
    ret = uv_udp_init_ex(ctx->loop, handle, flags);
                                            ^~~~~
src/udp.c:38:10: note: remove the 'if' if its condition is always true
    else if (lua_isstring(L, 1)) {
         ^~~~~~~~~~~~~~~~~~~~~~~~
src/udp.c:34:23: note: initialize the variable 'flags' to silence this warning
    unsigned int flags;
                      ^
                       = 0
```

Compilers don't know that after luaL_argerror(), the remaining code is unreachable. But it's better to initialize the flags to zero.

--- a/src/tcp.c
+++ b/src/tcp.c
@@ -32,7 +32,7 @@ static int luv_new_tcp(lua_State* L) {
     ret = uv_tcp_init(ctx->loop, handle);
   }
   else {
-    unsigned int flags;
+    unsigned int flags = AF_UNSPEC;
     if (lua_isnumber(L, 1)) {
       flags = lua_tointeger(L, 1);
     }

--- a/src/udp.c
+++ b/src/udp.c
@@ -31,7 +31,7 @@ static int luv_new_udp(lua_State* L) {
     ret = uv_udp_init(ctx->loop, handle);
   }
   else {
-    unsigned int flags;
+    unsigned int flags = AF_UNSPEC;
     if (lua_isnumber(L, 1)) {
       flags = lua_tointeger(L, 1);
     }

